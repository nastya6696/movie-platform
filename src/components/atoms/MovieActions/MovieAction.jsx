import CloseIcon from '../../../assets/svg/closeBtnPopup.svg';

import styles from './styles.module.scss';

const Actions = ['Edit', 'Delete']

export const MovieAction = ({isOpen, handleClose, handleActionClick, actions = Actions}) => {
  if(!isOpen) return null;

  return (
    <>
      <ul className={styles.actions} onClick={handleActionClick} >
        <div onClick={handleClose}>
          <img src={CloseIcon} alt="closeIcon" className={styles.closeIcon}/>
        </div>
        {actions.map(action => <li key={action}>{action}</li>)}
      </ul>
    </>
  )
}
