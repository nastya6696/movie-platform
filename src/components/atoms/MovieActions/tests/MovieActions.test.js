import { create } from 'react-test-renderer';
import {MovieAction} from "../MovieAction";

describe('MovieAction', () => {
  it('should create simple MovieAction components', () => {
    const component = create(
      <MovieAction
        isOpen
        handleActionClick={jest.fn()}
        handleClose={jest.fn()}
        actions={['Action1', 'Action2']}
      />);

    expect(component).toMatchSnapshot();
  });

  it('should not create simple MovieAction components', () => {
    const component = create(<MovieAction handleActionClick={jest.fn()} handleClose={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });
})
