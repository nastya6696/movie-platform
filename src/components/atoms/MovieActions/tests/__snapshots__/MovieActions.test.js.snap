// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieAction should create simple MovieAction components 1`] = `
<ul
  className="actions"
  onClick={[MockFunction]}
>
  <div
    onClick={[MockFunction]}
  >
    <img
      alt="closeIcon"
      className="closeIcon"
      src="closeBtnPopup.svg"
    />
  </div>
  <li>
    Action1
  </li>
  <li>
    Action2
  </li>
</ul>
`;

exports[`MovieAction should not create simple MovieAction components 1`] = `null`;
